packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -1886,13 +1886,20 @@ class _OverlayPortalState extends State<OverlayPortal> {
     if (oldWidget.controller != widget.controller) {
       oldWidget.controller._attachTarget = null;
       _setupController(widget.controller);
+    } else if (widget.controller._attachTarget == null) {
+      _setupController(widget.controller);
     }
   }
 
   @override
-  void dispose() {
-    assert(widget.controller._attachTarget == this);
+  void deactivate() {
```

Comment:
If the `OverlayPortal` is reparented, the `_attachTarget` has to be restored right? Also could you add a test for this?

Reply:
Sure

Reply:
I have pushed the required change.

Reply:
It should probably happen in `activate` instead of `didUpdateWidget`.

Reply:
Yaa, that will be correct. Let me do the change.